### PR TITLE
 Added testcase for backup pods verification

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
To verify if pxcentral-post-install-hook pod  is in completed state
**Which issue(s) this PR fixes** (optional)
Closes # PA-198, PA-199

**Special notes for your reviewer**:

This testcase verifies that the pxcentral-post-install-hook pod in the backup namespace is in Completed state.
Step: 
Get the list of pods in backup namespace
Filter the pxcentral-post-install-hook pod from the list and get the details of the pod
Verify if the reason for Terminated state of pod is " Completed"

### CODE REVIEW

File: tests/basic/backup_test.go [hunk 1]
```diff
@@ -1936,3 +1950,31 @@ func getBackupUID(orgID, backupName string) string {
...
+	ns := backup.GetPxBackupNamespace()
+	pods, err := core.Instance().GetPods(ns, labelSelectors)
+	if err != nil {
+		logrus.Warnf("Unable to fetch the pods from backup namespace\n Error : [%v]\n",
+			err)
+		return false
+	}
+	for _, pod := range pods.Items {
+		matched, _ := regexp.MatchString("^pxcentral-post-install-hook", pod.GetName())
+		if !matched {
```
Comment (vpinisetti-px) [inline tests/basic/backup_test.go:1965]: @snigdha-px you will have to validate whether pxcentral-post-install-hook pod is in completed state or not. This is the confirmation for px-backup successful installation.

**Review by vpinisetti-px — CHANGES_REQUESTED** [review]

File: tests/basic/backup_test.go [hunk 2]
```diff
@@ -1936,3 +1950,31 @@ func getBackupUID(orgID, backupName string) string {
 			orgID, backupName, err))
 	return backupUID
 }
+
+func validateBackupCluster() bool {
+	labelSelectors := make(map[string]string)
+	ns := backup.GetPxBackupNamespace()
+	pods, err := core.Instance().GetPods(ns, labelSelectors)
+	if err != nil {
+		logrus.Warnf("Unable to fetch the pods from backup namespace\n Error : [%v]\n",
```
Comment (kshithijiyer-px) [inline tests/basic/backup_test.go:1959]: Can we use an errorf here?

File: tests/basic/backup_test.go (hunk 1 above)
Comment (snigdha-px) [reply tests/basic/backup_test.go]: Done

File: tests/basic/backup_test.go (hunk 2 above)
Comment (snigdha-px) [reply tests/basic/backup_test.go]: Done

File: tests/basic/backup_test.go [hunk 3]
```diff
@@ -1956,24 +1956,28 @@ func validateBackupCluster() bool {
...
 	pods, err := core.Instance().GetPods(ns, labelSelectors)
 	if err != nil {
-		logrus.Warnf("Unable to fetch the pods from backup namespace\n Error : [%v]\n",
+		logrus.Errorf("Unable to fetch the pods from backup namespace\n Error : [%v]\n",
 			err)
 		return false
 	}
 	for _, pod := range pods.Items {
 		matched, _ := regexp.MatchString("^pxcentral-post-install-hook", pod.GetName())
-		if !matched {
```
Comment (vpinisetti-px) [inline tests/basic/backup_test.go]: @snigdha-px Imagine due to some issue in the product, other pods are not in running state but pxcentral-post-install-hook pod is in completed state. With this scenario, this logic doesn't work. Its better to keep code to validate all the pods status in backup cluster and verify the expected state based on pod.

**Review by vpinisetti-px — CHANGES_REQUESTED** [review]

File: tests/basic/backup_test.go (hunk 3 above)
Comment (snigdha-px) [reply tests/basic/backup_test.go]: Done

**Pull request closed without merging**